src/database/database.cpp

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <Poco/Data/SQLite/Connector.h>
+
+#include "database.hpp"
+
+using namespace Poco::Data::Keywords;
+
+namespace {
+enum CertColumns { eType = 0, eIssuer, eSerial, eCertURL, eKeyURL, eNotAfter };
+}
+
+/***********************************************************************************************************************
+ * Public
+ **********************************************************************************************************************/
+
+aos::Error Database::Init(const std::string& dbPath)
+{
+    if (mSession && mSession->isConnected()) {
+        return aos::ErrorEnum::eNone;
+    }
+
+    try {
+
+        Poco::Data::SQLite::Connector::registerConnector();
+        mSession = std::optional<Poco::Data::Session>(Poco::Data::Session("SQLite", dbPath));
+        CreateTables();
+
+    } catch (const Poco::Exception& e) {
+        return AOS_ERROR_WRAP(aos::ErrorEnum::eFailed);
+    }
+
+    return aos::ErrorEnum::eNone;
+}
+
+aos::Error Database::AddCertInfo(const aos::String& certType, const aos::iam::certhandler::CertInfo& certInfo)
+{
+    if (!mSession || !mSession->isConnected()) {
```

Comment:
It is not necessary to check these conditions every time.

Reply:
Remove